pkg/util/k8s/node.go

```diff
@@ -57,3 +59,63 @@ func GetNodeAddrs(node *v1.Node) (*ip.DualStackIPs, error) {
 	}
 	return nodeAddrs, nil
 }
+
+// GetNodeAddressFromAnnotations gets available IPs from the Node Annotation.The annotations are set by Antrea, including
+// NodeTransportAddressAnnotationKey string = "node.antrea.io/transport-addresses"
+func GetNodeAddressFromAnnotations(node *v1.Node, annotationKey string) (*ip.DualStackIPs, error) {
+	var ipAddrs = new(ip.DualStackIPs)
+	annotationAddrsStr := node.Annotations[annotationKey]
+	if annotationAddrsStr != "" {
+		for _, addr := range strings.Split(annotationAddrsStr, ",") {
+			peerNodeAddr := net.ParseIP(addr)
+			if peerNodeAddr == nil {
+				return nil, fmt.Errorf("invalid annotation for ip-address on Node %s: %s", node.Name, annotationAddrsStr)
+			}
+			if peerNodeAddr.To4() == nil {
+				ipAddrs.IPv6 = peerNodeAddr
+			} else {
+				ipAddrs.IPv4 = peerNodeAddr
+			}
+		}
+		return ipAddrs, nil
+	}
+	return nil, nil
+}
+
+// GetNodeGWIPs gets Node Antrea gateway IPs from the Node Spec.
+func GetNodeGWIPs(node *v1.Node) (*ip.DualStackIPs, error) {
+	getIPFromPodCIDR := func(podCIDR string) (*net.IPNet, error) {
+		_, localSubnet, err := net.ParseCIDR(podCIDR)
```

Comment:
It doesn't make sense to name the variable "localSubnet" in this context.

Reply:
Done.